trismart_smartboard_integration/models/smartboard_connector.py

```diff
@@ -5,44 +5,58 @@ class SmartBoard(models.Model):
-                    if partner_id:
-                        customer = self.env['res.partner'].browse(partner_id[0])
+            if sb_lead_id:
+                if type(sb_lead_id) == int:
+                    partner_exist = self.env['res.partner'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+                    lead_exist = self.env['crm.lead'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+                    project_exist = self.env['project.project'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+
+                    if not partner_exist:
+                        # Create new partner
+                        partner_new = self.env['res.partner'].create({'name': 'SBLead-' + str(sb_lead_id), 'sb_lead_id': sb_lead_id, "sync_status" : '1'})
+
+                    if not lead_exist:
+                        # Create new lead
+                        lead_new = self.env['crm.lead'].create({'name': 'SBLead-' + str(sb_lead_id), 'sb_lead_id': sb_lead_id})
+                        # Convert lead to opportunity
+                        if partner_exist:
+                            vals = lead_new[0]._convert_opportunity_data(partner_exist, team_id=False)
+                        else:
+                            vals = lead_new[0]._convert_opportunity_data(partner_new, team_id=False)
+                        lead_new.write(vals)
+
+                    if not project_exist:
+                        project_env = self.env['project.project']
+                        # Create new project
+                        if project_template_id: # Template ID is provided
+                            if type(project_template_id) == int:
+                                template = project_env.search([('id', '=', project_template_id), ('name', 'like', ' (TEMPLATE)')])
+                                if template: # Template exists
+                                    project = template.create_project_from_template(sb_lead_id)
+                                    project_new = project_env.search([('id', '=', project['res_id'])], limit=1)
+                                else: # Template does not exist, treat as no template ID
+                                    project_new = project_env.create({"name": 'SBLead-' + str(sb_lead_id), "sb_lead_id": sb_lead_id})
+                            else: # Template ID is not an Integer, treat as no template ID
+                                project_new = project_env.create({"name": 'SBLead-' + str(sb_lead_id), "sb_lead_id": sb_lead_id})
+                        else: # Template ID is not provided
+                            project_new = project_env.create({"name": 'SBLead-' + str(sb_lead_id), "sb_lead_id": sb_lead_id})
+                        # Add partner_id
+                        if partner_exist:
+                            project_new.partner_id = partner_exist.id
+                        else:
+                            project_new.partner_id = partner_new.id
```

Comment:
Should we do this:  partner_to_assign = (partner_exist or partner_new) to reduce the if-else in the code?

Reply:
resolved in v4